pkg/yang/entry.go

```diff
@@ -1248,6 +1252,37 @@ func (e *Entry) ApplyDeviate(deviateOpts ...DeviateOpt) []error {
 						deviatedNode.ListAttr.MaxElements = math.MaxUint64
 					}
 
+					if musts, exists := devSpec.Extra["must"]; exists {
+						// range through the deviations to apply them
+						for _, mustDeleteInterf := range musts {
+
+							mustDelete := mustDeleteInterf.(*Must)
+
+							found := false
+							var idx int
+							var existingMustInterf interface{}
+							var existingMust *Must
+							for idx, existingMustInterf = range deviatedNode.Extra["must"] {
+								if existingMustInterf.(*Must).Name == mustDelete.Name {
```

Comment:
It might be worth pulling this logic out into a helper function that states what it is doing a bit more clearly -- for example, something like `isSameMustStmt`, given that I assume that we actually need to handle other kind of deviations for `must` statements, right? (did not check rfc7950 here)

Reply:
Changed the implementation meanwhile.
My take in the RFC was that all elements must match, but after some discussions seems like only the name (the actual must expression) must match, the error code etc does not matter. Hence the code is a lot simpler.
Change pushed